File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/CharacterClassAbbrev.pl
#!/usr/bin/perl -w                                          
  
## CharacterClassAbbrev.pl

use strict;                                                           

my $pattern = '\b(\d\d:\d\d:\d\d)\b';                               #(A)
my $input_string = "The game starts at 15:30:00\n";                 #(B)
printf "Start time: %s\n", $input_string =~ $pattern;               #(C)
                       # Start time: 15:30:00

$pattern = '(\w\w\w\w)';                                            #(D)
$input_string = "abracadabra is a magical chant";                   #(E)
my @matched_substrings = $input_string =~ /$pattern/g;              #(F)
print "@matched_substrings\n";                                      #(G)
                       # abra cada magi chan

my @words = split /[\s,-]/, "apples,oranges mangoes pixie-fruit";   #(H)
print "@words\n";                                                   #(I)   
                       # apples oranges mangoes pixie fruit

@words = split /[\W]/, "apples,oranges mangoes pixie-fruit";        #(J)
print "@words\n";                                                   #(K)
                       # apples oranges mangoes pixie fruit

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/CharacterClassAbbrev.py
#!/usr/bin/python
  
### CharacterClassAbbrev.py

import re                                                           #(A)

pattern = r'\b(\d\d:\d\d:\d\d)\b'                                   #(B)
input_string = "The game starts at 15:30:00\n"                      #(C)
m = re.search( pattern, input_string )                              #(D)
if ( m != None ):                                                   #(E)
    print "Start time: %s" % m.group(1)                             #(F)
               # Start time: 15:30:00

pattern = r'(\w\w\w\w)'                                             #(G)
input_string = "abracadabra is a magical chant"                     #(H)
matched_substrings = re.findall( pattern, input_string )            #(I)
if ( matched_substrings != None ):                                  #(J)
    print matched_substrings                                        #(K)
               # ['abra', 'cada', 'magi', 'chan']

words = re.split( r'[\s,-]', "apples,oranges mangoes pixie-fruit" ) #(L)
print words                                                         #(M)
               # ['apples', 'oranges', 'mangoes', 'pixie', 'fruit']

words = re.split( r'[\W]', "apples,oranges mangoes pixie-fruit")    #(N)
print words                                                         #(O)
               # ['apples', 'oranges', 'mangoes', 'pixie', 'fruit']
